lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -198,4 +204,62 @@ class FlDotData {
     this.dotSize = 4.0,
     this.checkToShowDot = showAllDots,
   });
-}
\ No newline at end of file
+}
+
+class DashDefinition {
+  final double solidWidth;
+  final double gapWidth;
+
+  const DashDefinition({this.solidWidth, this.gapWidth});
+}
+
+class LineChartLine extends FlLine {
+  final DashDefinition dashDefinition;
+
+  const LineChartLine({
+    Color color,
+    double strokeWidth,
+    this.dashDefinition,
+  }) : super(color: color, strokeWidth: strokeWidth);
+}
+
+class VerticalLine extends LineChartLine {
+  final double x;
+  final double endY;
```

Comment:
we don't need endY because ExtraLines should be drawn from y = 0 to y = height, and the only x is dynamic ( for the below line we should apply another solution), prefer this ExtraLines just for avg line.

Reply:
Yeah but what if the user wants the line to end at a data point for example?

Reply:
If we want to draw this to the specific data point then we should have start data point alongside it.
I said, use it just for full horizontal and vertical lines, and it can be used as a avg line in your usecase, then you should use BelowBarData to draw lines below dot points

Reply:
After checking it some more I understand better what below bar data is for, so I'm dropping endY.